Compressor.cu

```diff
@@ -18,6 +18,8 @@
 
 #include "gpuHuffmanConstruction.h"
 
+#define BLOCK_SIZE 256
```

Comment:
Could you use a template instead of a macro? I think this will introduce some conflicts because there are multiple kernels in this file.